### DISCUSSION TITLE
Need suggestions on training vanna with enums.

### DISCUSSION THREAD

**Discussion by kernelism** [vanna-ai/vanna#276] [Q&A] [answered]
So I can't find a neat way to give the llm context on the enums.
I could do it as documentation, but that seems a bit clumsy. 
The information schema or DDL commands alone are not enough to convey this.
Can anyone help me out. I feel like I'm missing something.

**Comment by zainhoda** [comment, marked as answer]
For now documentation is the way to do this. I agree that it's clumsy but there's a reason for it.

This was a legacy choice made by us 6 months ago when people didn't have trusted ways to run an LLM so we used to make an explicit choice that the Vanna package would never automatically crawl the contents of a database. We made a conscious decision to not feed query results back into the LLM responses, which is what would be necessary to achieve what you're talking about.

However, based on feedback now that people trust LLMs more in general and/or have ways of running LLMs in a trusted manner, we are incrementally adding functionality like this that will allow the system to introspect the database as necessary. We will always put this behind an explicit flag (`allow_llm_to_see_data=True`) that will be opt-in.

In the near-ish future we will make it so that we can do multi-stage queries where a question could first produce an introspection-type query to find out the distinct values of a particular field before proceeding to write the "actual" query that would give the response.

**Reply by cta2106** [reply]
@zainhoda thanks for this! Do you have a timeline on when the multi-stage querying might be ready?

**Comment by kernelism** [comment]
Hey, thanks for the response. It makes quite sense.
Could you elaborate on what exactly `allow_llm_to_see_data = True` does?
Simply lets the llm see the query results?

**Reply by zainhoda** [reply]
Yeah, it’s just a setting that we’ll use in places like the Flask app that will explicitly enable the functions that require the dataframes to be passed to the LLM
